File: lab-2.2-ByLiza/C#/BaseString.cs
namespace String
{
    public class BaseString
    {
        protected string _value; 

        public string Value { get => _value; } 

        public BaseString(string input)
        {
            _value = input;
        }

        public virtual int GetLength()
        {
            return _value.Length;
        } 

        public virtual string Shift()
        {
            return _value;
        }
    }
}

File: lab-2.2-ByLiza/C++/Basestring.cpp
#include "BaseString.h"

namespace String {

    BaseString::BaseString(const std::string& input) : _value(input) {}

    std::string BaseString::GetValue() const {
        return _value;
    }

    int BaseString::GetLength() const {
        return _value.length();
    }

    std::string BaseString::Shift() const {
        return _value;
    }

}